chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -16,7 +16,8 @@ import scala.concurrent.{ExecutionContext, Future}
   */
 case class ChainHandler(
     blockHeaderDAO: BlockHeaderDAO,
```

Comment:
Chances are you're aware of this, but IMO one goal here should be to remove this parameter from the class, and then give it a method `persistToDisk(dao)` (or something similar) that writes once